src/ripple/ledger/impl/RawStateTable.cpp

```diff
@@ -354,5 +354,18 @@ RawStateTable::slesEnd (ReadView const& base) const
             base.sles.end(), base.sles.end());
 }
 
+std::unique_ptr<ReadView::sles_type::iter_base>
+RawStateTable::slesUpperBound (ReadView const& base, uint256 const& key) const
+{
+    using iterator = ReadView::sles_type::iterator;
+    return std::upper_bound(
+        iterator(&base, slesBegin(base)),
+        iterator(&base, slesEnd(base)),
+        key, [](uint256 const& x, iterator::value_type const& y)
+             {
+                  return x < y->key();
+             }).extract();
+}
```

Comment:
Something ain't right here, `std::upper_bound` complexity is linear for non-RandomAccessIterators... this would kill performance for resuming on a marker.